doc/manual.rst

```diff
@@ -2975,6 +2975,21 @@ Even some code that has side effects is permitted in a static block:
   static:
     echo "echo at compile time"
 
+When `static` is used like a procedure call with a proc call as an argument,
+that proc call gets executed at compile-time.
+
+.. code-block:: nim
+
+  proc getNum(a: int): int = a
+
+  # Below calls "echo getNum(123)" at compile time.
+  static:
+    echo getNum(123)
+
+  # Below call evaluates the "getNum(123)" at compile time, but its
+  # result gets used at run time.
+  echo getNum(123).static
```

Comment:
```suggestion
  # The following expressions are equivalent:

  static:
    getNum(123)

  getNum(123).static

  static(getNum(123))
```
I think it makes more sense to show that using `static` like a routine is equivalent to using a `static:` block.

Reply:
This change will mislead the readers just as it misled me [here](https://github.com/nim-lang/Nim/issues/16987#issuecomment-955853383).

I intentionally did not put all the callable style in that section of the manual as we already have a section to deal with UFCS.

Reply:
IMO we shouldn't document that `x.static` works, it's an accidental feature.

Reply:
Do you mean that only the `x.static` style working is accidental, or everything documented in the test in this PR (e.g. `static(x)`?

Can you update the original issue this PR is attempting to close?

Reply:
Both `x.static` and `x.static()` are not supposed to work, `static(x)` should be documented.

Reply:
@Araq Done. Please look at the latest collective diff of this PR.

Reply:
> Both `x.static` and `x.static()` are not supposed to work, `static(x)` should be documented.

@Araq What's the reasoning behind this? It works for "normal" function call syntax, so why not for MCS as well?

Reply:
I should have worded this better: It's not supposed to "not work" either, it's just that I don't know the consequences of allowing it. We already have restrictions on the method call syntax for technical reasons and this one feels like it could be affected too in alternative/future implementations.